src/components/settings/SettingsAppearanceTab/CustomThemeModal.jsx

```diff
@@ -0,0 +1,137 @@
+    close();
+  }
+
+  return (
+    <ModalDialog
+      title="My dialog"
+      isOpen={isOpen}
+      onClose={close}
+      size="md"
+      hasCloseButton
+      isFullscreenOnMobile
+    >
+      <ModalDialog.Header>
+        <ModalDialog.Title>
+          Custom Theme
+        </ModalDialog.Title>
+        <p className="mt-3">{headerText}</p>
+      </ModalDialog.Header>
+
+      <ModalDialog.Body>
+        <Form>
+          <Form.Group controlId="custom-button">
+            <Form.Control
+              floatingLabel="Button color"
+              isInvalid={!buttonValid}
+              onChange={(e) => {
+                setButton(e.target.value);
+                validateColor(e.target.value, setButtonValid);
+              }}
+              defaultValue={customColors?.button}
+            />
+            {!buttonValid && (<Form.Control.Feedback type="invalid">{invalidMessage}</Form.Control.Feedback>)}
+          </Form.Group>
+          <Form.Group controlId="custom-banner">
+            <Form.Control
+              floatingLabel="Banner color"
+              isInvalid={!bannerValid}
+              onChange={(e) => {
+                setBanner(e.target.value);
+                validateColor(e.target.value, setBannerValid);
+              }}
+              defaultValue={customColors?.banner}
+            />
+            {!bannerValid && (<Form.Control.Feedback type="invalid">{invalidMessage}</Form.Control.Feedback>)}
+          </Form.Group>
+          <Form.Group controlId="custom-accent">
+            <Form.Control
+              floatingLabel="Accent color"
+              isInvalid={!accentValid}
+              onChange={(e) => {
+                setAccent(e.target.value);
+                validateColor(e.target.value, setAccentValid);
+              }}
+              defaultValue={customColors?.accent}
+            />
+            {!accentValid && (<Form.Control.Feedback type="invalid">{invalidMessage}</Form.Control.Feedback>)}
+          </Form.Group>
+        </Form>
+        <p className="mt-4 mb-1 d-flex x-small">
+          <Icon src={InfoOutline} className="mr-2" /> {infoText}
+          <Hyperlink
+            destination={HELP_CENTER_BRANDING_LINK}
+            style={{ display: 'contents' }}
+            target="_blank"
+          >
+            Enterprise Help Center.
+          </Hyperlink>
+        </p>
+      </ModalDialog.Body>
+
+      <ModalDialog.Footer>
+        <ActionRow>
+          <Button
+            variant="primary"
+            onClick={setCustom}
+            disabled={!buttonValid || !bannerValid || !accentValid}
+          >
+            Add theme
+          </Button>
+        </ActionRow>
+      </ModalDialog.Footer>
+    </ModalDialog>
+
+  );
+}
+
+CustomThemeModal.defaultProps = {
+  customColors: null,
+};
+
+CustomThemeModal.propTypes = {
+  isOpen: PropTypes.bool.isRequired,
+  close: PropTypes.func.isRequired,
+  customColors: PropTypes.arrayOf(
+    PropTypes.strings,
```

Comment:
nit: Is `PropTypes.strings` a valid property? I haven't seen it plural before, only `PropTypes.string` 🤔